syntax = "proto3";
package Q2;

message student{
    string name = 1;
    int32 rollNum = 2;
    int32 class = 3;
    string sec = 4;

    enum guide{
        PARENT = 0;
        GUARDIAN = 1;
    }
    enum parents{
        FATHER = 0;
        MOTHER = 1;
    }
    message guideBy{
        string number = 1;
        string name = 2;
        guide gType = 3;
        parents pType = 4;
    }
    message semesterDetails{
        optional int32 number = 1;
        message Marks{
            optional string subjectName = 1;
            optional int32 score = 2;
        }
        repeated Marks marks = 2;
    }
    repeated guideBy guideby= 5;
    optional semesterDetails semesterdetails = 6;
}

message allStudents{
    repeated student allstudent = 1;
}